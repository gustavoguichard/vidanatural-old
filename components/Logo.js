import { memo } from 'react'
import { scrollToId } from 'utils/helpers'
import ImageReplace from 'components/ImageReplace'

const Logo = ({ clickable = false, onClick, style, ...props }) => (
  <ImageReplace
    src="logo-white.png"
    onClick={
      clickable
        ? () => {
            onClick && onClick()
            scrollToId()
          }
        : undefined
    }
    width="130"
    height="117"
    className="logo"
    style={{ cursor: clickable ? 'pointer' : 'inherit', ...style }}
    {...props}
  >
    <h1>Vida Natural</h1>
  </ImageReplace>
)

export default memo(Logo)
