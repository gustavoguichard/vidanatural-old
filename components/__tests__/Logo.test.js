import { render } from 'react-testing-library'
import Logo from 'components/Logo'

it('renders correctly', () => {
  const { container } = render(<Logo style={{ margin: 0 }} foo="bar" />)
  expect(container).toMatchSnapshot()
})
