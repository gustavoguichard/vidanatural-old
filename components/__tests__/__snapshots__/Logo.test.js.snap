// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="logo"
    foo="bar"
    style="display: block; background: url(/static/logo-white.png) no-repeat center; background-size: contain; height: 117px; width: 130px; overflow: hidden; text-indent: -999em; cursor: inherit; margin: 0px;"
  >
    <h1>
      Vida Natural
    </h1>
  </div>
</div>
`;
